src/libostree/ostree-sysroot-cleanup.c

```diff
@@ -501,8 +552,22 @@ _ostree_sysroot_cleanup_internal (OstreeSysroot              *self,
 
   if (do_prune_repo)
     {
-      if (!prune_repo (repo, cancellable, error))
-        return glnx_prefix_error (error, "Pruning repo");
+      gint n_objects_total;
+      gint n_objects_pruned;
+      guint64 freed_space;
+      g_autoptr(GHashTable) reachable = ostree_repo_traverse_new_reachable ();
+      OstreeRepoPruneOptions opts = { OSTREE_REPO_PRUNE_FLAGS_REFS_ONLY, reachable };
+      if (!ostree_sysroot_cleanup_prune_repo (self, &opts, &n_objects_total,
+                                              &n_objects_pruned, &freed_space,
+                                              cancellable, error))
+        return FALSE;
+
+      /* TODO remove printf in library */
+      if (freed_space > 0)
+        {
+          g_autofree char *freed_space_str = g_format_size_full (freed_space, 0);
+          g_print ("Freed objects: %s\n", freed_space_str);
```

Comment:
Wait, but the commit message is implying we're no longer doing prints, no? Or does it actually mean we have the API in place now so that clients can workaround this?

Reply:
This is the old method; the new API indeed does not print.